Migrate from yarn to npm Issue #3511

EnTRoPY0120 opened the pull request:
- [x] Removed  ```yarn.lock and .yarnrc```
- [x] Added ```package-lock.json```
- [x] Replace ```yarn``` with ```npm``` in ```package.json```
- [x] Refactor ```yarn``` with ```npm``` in Documentation,```Dockerfile``` and Github actions
- [x] Refactor ```yarn``` with ```npm``` in ```git-hooks/pre-commit```.

P.S:
There's a couple of deprecated packages with vulnerabilities. If you want I can look into them.

dtslvr commented on README.md:
```diff
@@ -150,15 +150,15 @@ Ghostfolio is available for various home server systems, including [CasaOS](http
 - [Docker](https://www.docker.com/products/docker-desktop)
 - [Node.js](https://nodejs.org/en/download) (version 20+)
-- [Yarn](https://yarnpkg.com/en/docs/install)
+- [npm](https://docs.npmjs.com/cli/v10/commands/npm-install)
```
Is `npm` explicitly needed or is it already part of `Node.js`?

EnTRoPY0120 replied:
I'm sorry I thought of including nvm, but added npm instead,my bad. Yes, it's already part of nodejs.

dtslvr commented on package.json:
I prefer to have fixed versions (without `^`). Could you please adapt them to the newest possible ones?

EnTRoPY0120 replied:
Sure, I can do that.

dtslvr commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM --platform=$BUILDPLATFORM node:20-slim as builder
+FROM --platform=$BUILDPLATFORM node:20-slim AS builder
```
👍🏻

dtslvr commented on Dockerfile:
```diff
@@ -8,18 +8,17 @@ WORKDIR /ghostfolio
-    python3 \
-    && rm -rf /var/lib/apt/lists/*
-RUN yarn install
+RUN apt-get update && apt-get install -y \
```
Why `apt-get` instead of `apt`?

EnTRoPY0120 replied:
Both of can be used but from what I have seen in popular images on Dockerhub and Docker usually prefer apt-get since it's older and stable. apt is newer and generally stable. If you want I can leave it as it is as apt install.

dtslvr commented on README.md:
```diff
@@ -168,31 +168,31 @@ Ghostfolio is available for various home server systems, including [CasaOS](http
 #### Serve
 
-Run `yarn start:server`
+Run `npm start:server`
```
`npm run start:server`❓

EnTRoPY0120 replied:
There is a separate script to start server,client and storybook for local development. Am I missing something?. I ran it this way as well.

dtslvr replied:
I have added **run** 🙂

dtslvr commented on README.md:
```diff
@@ -168,31 +168,31 @@ Ghostfolio is available for various home server systems, including [CasaOS](http
 ### Start Client
 
-Run `yarn start:client` and open https://localhost:4200/en in your browser
+Run `npm start:client` and open https://localhost:4200/en in your browser
```
`npm run start:client`❓

EnTRoPY0120 replied:
Sorry I forgot to add run. That was really dumb from my side.

dtslvr commented on README.md:
```diff
@@ -168,31 +168,31 @@ Ghostfolio is available for various home server systems, including [CasaOS](http
 ### Start _Storybook_
 
-Run `yarn start:storybook`
+Run `npm start:storybook`
```
`npm run start:storybook`❓

dtslvr reviewed (changes requested):
Thanks for your contribution, @EnTRoPY0120! The build is passing, which is already nice 😎 I have added a few comments.

EnTRoPY0120 commented:
> Thanks for your contribution, @EnTRoPY0120! The build is passing, which is already nice 😎 I have added a few comments.

@dtslvr Thanks and I have made the changes and updates the pull request, verify if it's okay or if you want any other changes.

dtslvr commented:
I have rebased this branch with `main` and now the build is not working anymore. When I run `npm install` locally, I get the following error:

```
ghostfolio % npm install
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: @angular-eslint/eslint-plugin@18.1.0
npm error Found: @typescript-eslint/utils@6.21.0
npm error node_modules/@typescript-eslint/utils
npm error   @typescript-eslint/utils@"6.21.0" from @typescript-eslint/eslint-plugin@6.21.0
npm error   node_modules/@typescript-eslint/eslint-plugin
npm error     dev @typescript-eslint/eslint-plugin@"6.21.0" from the root project
npm error   @typescript-eslint/utils@"6.21.0" from @typescript-eslint/type-utils@6.21.0
npm error   node_modules/@typescript-eslint/eslint-plugin/node_modules/@typescript-eslint/type-utils
npm error     @typescript-eslint/type-utils@"6.21.0" from @typescript-eslint/eslint-plugin@6.21.0
npm error     node_modules/@typescript-eslint/eslint-plugin
npm error       dev @typescript-eslint/eslint-plugin@"6.21.0" from the root project
npm error
npm error Could not resolve dependency:
npm error peer @typescript-eslint/utils@"^7.11.0 || ^8.0.0-alpha.37" from @angular-eslint/eslint-plugin@18.1.0
npm error node_modules/@angular-eslint/eslint-plugin
npm error   dev @angular-eslint/eslint-plugin@"18.1.0" from the root project
npm error
npm error Conflicting peer dependency: @typescript-eslint/utils@7.17.0
npm error node_modules/@typescript-eslint/utils
npm error   peer @typescript-eslint/utils@"^7.11.0 || ^8.0.0-alpha.37" from @angular-eslint/eslint-plugin@18.1.0
npm error   node_modules/@angular-eslint/eslint-plugin
npm error     dev @angular-eslint/eslint-plugin@"18.1.0" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

Any idea how to fix this @EnTRoPY0120?

EnTRoPY0120 commented:
> I have rebased this branch with `main` and now the build is not working anymore. When I run `npm install` locally, I get the following error:
> 
> ```
> ghostfolio % npm install
> npm error code ERESOLVE
> npm error ERESOLVE could not resolve
> npm error
> npm error While resolving: @angular-eslint/eslint-plugin@18.1.0
> npm error Found: @typescript-eslint/utils@6.21.0
> npm error node_modules/@typescript-eslint/utils
> npm error   @typescript-eslint/utils@"6.21.0" from @typescript-eslint/eslint-plugin@6.21.0
> npm error   node_modules/@typescript-eslint/eslint-plugin
> npm error     dev @typescript-eslint/eslint-plugin@"6.21.0" from the root project
> npm error   @typescript-eslint/utils@"6.21.0" from @typescript-eslint/type-utils@6.21.0
> npm error   node_modules/@typescript-eslint/eslint-plugin/node_modules/@typescript-eslint/type-utils
> npm error     @typescript-eslint/type-utils@"6.21.0" from @typescript-eslint/eslint-plugin@6.21.0
> npm error     node_modules/@typescript-eslint/eslint-plugin
> npm error       dev @typescript-eslint/eslint-plugin@"6.21.0" from the root project
> npm error
> npm error Could not resolve dependency:
> npm error peer @typescript-eslint/utils@"^7.11.0 || ^8.0.0-alpha.37" from @angular-eslint/eslint-plugin@18.1.0
> npm error node_modules/@angular-eslint/eslint-plugin
> npm error   dev @angular-eslint/eslint-plugin@"18.1.0" from the root project
> npm error
> npm error Conflicting peer dependency: @typescript-eslint/utils@7.17.0
> npm error node_modules/@typescript-eslint/utils
> npm error   peer @typescript-eslint/utils@"^7.11.0 || ^8.0.0-alpha.37" from @angular-eslint/eslint-plugin@18.1.0
> npm error   node_modules/@angular-eslint/eslint-plugin
> npm error     dev @angular-eslint/eslint-plugin@"18.1.0" from the root project
> npm error
> npm error Fix the upstream dependency conflict, or retry
> npm error this command with --force or --legacy-peer-deps
> npm error to accept an incorrect (and potentially broken) dependency resolution.
> ```
> 
> Any idea how to fix this @EnTRoPY0120?

I removed the package-lock.json and node modules from those last commits since I had changed package.json since you asked me to change those packages to fixed versions and ran npm install. It ran successfully.

dtslvr commented:
Any idea how to fix the current issue @EnTRoPY0120?

https://github.com/ghostfolio/ghostfolio/pull/3601#issuecomment-2251096618

EnTRoPY0120 commented:
I have updated the package-lock.json and package.json with some latest versions. It ran successfully. Check if the latest changes fix the dependency issue at your end @dtslvr?

dtslvr reviewed (approved):
After deleting both `node_modules` AND `package-lock.json` the build is finally passing. 

Thank you for your contribution, @EnTRoPY0120! 🙌🏻 As well as thanks to @DanielVenable for the groundwork 🙂

Ready to merge now.

EnTRoPY0120 commented:
I wanted to contribute since I really liked the app. I wish I could contribute more in the future too. Thanks for the guidance as well :) @dtslvr.
